src/openforms/prefill/contrib/stufbg/plugin.py

```diff
@@ -119,6 +119,13 @@ def _get_values_for_bsn(
             if value and "@noValue" not in value:
                 response_dict[attribute] = value
 
+        # postcodes in StUF BG responses have this form "[1-9][0-9]{3}[A-Z]{0,2}"
+        # Our prefill tests expect roughly "[1-9][0-9]{3} [A-Z]{0,2}"
+        if FieldChoices.postcode in response_dict and " " not in (
+            postcode := response_dict[FieldChoices.postcode]
+        ):
+            response_dict[FieldChoices.postcode] = postcode[:4] + " " + postcode[4:]
```

Comment:
There is a bit more nuance to this - I have no idea if it conflicts or not, but we run normalizers for the Formio data fields - see `openforms.formio.components.custom.Postcode.normalizer`

Though that also makes me think - we deprecated the custom postcode component in favour of a pre-configured text field, and that one probably does not run this normalizer. The functionality there is generic enough to conform the value to the mask. I'll have to create a ticket!